### ISSUE TITLE
Install issue: `fatal: repository 'https://github.com/debris/bignumber.js.git/' not found`

### ISSUE THREAD

**Issue by mrosendin** [ConsenSys-archive/ganache-cli-archive#332]
I'm trying to install with `npm install -g ethereumjs-testrpc`.

## Current Behavior
Terminal output:
```
❯ npm install -g ethereumjs-testrpc
npm ERR! git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: Cloning into bare repository '/Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
npm ERR! git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: remote: Repository not found.
npm ERR! git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: fatal: repository 'https://github.com/debris/bignumber.js.git/' not found
npm ERR! code 128
npm ERR! Command failed: git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1
npm ERR! Cloning into bare repository '/Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
npm ERR! remote: Repository not found.
npm ERR! fatal: repository 'https://github.com/debris/bignumber.js.git/' not found
npm ERR!

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/matthew/.npm/_logs/2017-06-23T22_54_11_576Z-debug.log
```

Truncated log file (`2017-06-23T22_54_11_576Z-debug.log`) output:

```
...
13762 silly mapToRegistry uri https://registry.npmjs.org/xmlhttprequest
13763 verbose addNameTag registry:https://registry.npmjs.org/xmlhttprequest already in flight; waiting
13764 silly addNameTag next cb for xmlhttprequest with tag latest
13765 silly addNamed xmlhttprequest@1.8.0
13766 verbose addNamed "1.8.0" is a plain semver version for xmlhttprequest
13767 silly cache afterAdd crypto-js@3.1.8
13768 verbose afterAdd /Users/matthew/.npm/crypto-js/3.1.8/package/package.json not in flight; writing
13769 verbose correctMkdir /Users/matthew/.npm correctMkdir not in flight; initializing
13770 silly cache afterAdd xmlhttprequest@1.8.0
13771 verbose afterAdd /Users/matthew/.npm/xmlhttprequest/1.8.0/package/package.json not in flight; writing
13772 verbose correctMkdir /Users/matthew/.npm correctMkdir not in flight; initializing
13773 verbose afterAdd /Users/matthew/.npm/crypto-js/3.1.8/package/package.json written
13774 verbose afterAdd /Users/matthew/.npm/xmlhttprequest/1.8.0/package/package.json written
13775 error git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: Cloning into bare repository '/Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
13775 error git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: remote: Repository not found.
13775 error git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: fatal: repository 'https://github.com/debris/bignumber.js.git/' not found
13776 silly fetchPackageMetaData Error: Command failed: git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1
13776 silly fetchPackageMetaData Cloning into bare repository '/Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
13776 silly fetchPackageMetaData remote: Repository not found.
13776 silly fetchPackageMetaData fatal: repository 'https://github.com/debris/bignumber.js.git/' not found
13776 silly fetchPackageMetaData
13776 silly fetchPackageMetaData     at ChildProcess.exithandler (child_process.js:205:12)
13776 silly fetchPackageMetaData     at emitTwo (events.js:106:13)
13776 silly fetchPackageMetaData     at ChildProcess.emit (events.js:194:7)
13776 silly fetchPackageMetaData     at maybeClose (internal/child_process.js:899:16)
13776 silly fetchPackageMetaData     at Socket.<anonymous> (internal/child_process.js:342:11)
13776 silly fetchPackageMetaData     at emitOne (events.js:96:13)
13776 silly fetchPackageMetaData     at Socket.emit (events.js:191:7)
13776 silly fetchPackageMetaData     at Pipe._handle.close [as _onclose] (net.js:510:12)
13776 silly fetchPackageMetaData  error for bignumber.js@git+https://github.com/debris/bignumber.js.git#master { Error: Command failed: git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1
13776 silly fetchPackageMetaData Cloning into bare repository '/Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
13776 silly fetchPackageMetaData remote: Repository not found.
13776 silly fetchPackageMetaData fatal: repository 'https://github.com/debris/bignumber.js.git/' not found
13776 silly fetchPackageMetaData
13776 silly fetchPackageMetaData     at ChildProcess.exithandler (child_process.js:205:12)
13776 silly fetchPackageMetaData     at emitTwo (events.js:106:13)
13776 silly fetchPackageMetaData     at ChildProcess.emit (events.js:194:7)
13776 silly fetchPackageMetaData     at maybeClose (internal/child_process.js:899:16)
13776 silly fetchPackageMetaData     at Socket.<anonymous> (internal/child_process.js:342:11)
13776 silly fetchPackageMetaData     at emitOne (events.js:96:13)
13776 silly fetchPackageMetaData     at Socket.emit (events.js:191:7)
13776 silly fetchPackageMetaData     at Pipe._handle.close [as _onclose] (net.js:510:12)
13776 silly fetchPackageMetaData   killed: false,
13776 silly fetchPackageMetaData   code: 128,
13776 silly fetchPackageMetaData   signal: null,
13776 silly fetchPackageMetaData   cmd: 'git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1' }
13777 silly timing Tried to end timer that doesn't exist: stage:runTopLevelLifecycles
13778 silly install printInstalled
13779 verbose stack Error: Command failed: git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1
13779 verbose stack Cloning into bare repository '/Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
13779 verbose stack remote: Repository not found.
13779 verbose stack fatal: repository 'https://github.com/debris/bignumber.js.git/' not found
13779 verbose stack
13779 verbose stack     at ChildProcess.exithandler (child_process.js:205:12)
13779 verbose stack     at emitTwo (events.js:106:13)
13779 verbose stack     at ChildProcess.emit (events.js:194:7)
13779 verbose stack     at maybeClose (internal/child_process.js:899:16)
13779 verbose stack     at Socket.<anonymous> (internal/child_process.js:342:11)
13779 verbose stack     at emitOne (events.js:96:13)
13779 verbose stack     at Socket.emit (events.js:191:7)
13779 verbose stack     at Pipe._handle.close [as _onclose] (net.js:510:12)
13780 verbose cwd /Users/matthew/.npm/_logs
13781 verbose Darwin 16.6.0
13782 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g" "ethereumjs-testrpc"
13783 verbose node v7.9.0
13784 verbose npm  v4.6.1
13785 error code 128
13786 error Command failed: git clone --template=/Users/matthew/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1
13786 error Cloning into bare repository '/Users/matthew/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
13786 error remote: Repository not found.
13786 error fatal: repository 'https://github.com/debris/bignumber.js.git/' not found
13787 verbose exit [ 1, true ]
```

* Version used: latest (3.0.5)
* Node version: v7.9.0
* Operating System and version: macOS Sierra 10.12.5
* NPM version 4.6.1 (also tried v5.3)

**Comment by rneild** [comment]
I see the same.

**Comment by danchay** [comment]
Yep.

npm install -g ethereumjs-testrpc
npm ERR! git fetch -a origin (https://github.com/debris/bignumber.js.git) remote: Repository not found.
npm ERR! git fetch -a origin (https://github.com/debris/bignumber.js.git) fatal: repository 'https://github.com/debris/bignumber.js.git/' not found

Same dependency issue with web3.js:
https://github.com/ethereum/web3.js/pull/905

**Comment by OR13** [comment]
Yep, this is broken for me as well:

Here is a travis failure: https://travis-ci.org/transmute-industries/ico/builds/246456749

```
npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /home/travis/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: Cloning into bare repository '/home/travis/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /home/travis/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: remote: Invalid username or password.
npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /home/travis/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1: fatal: Authentication failed for 'https://github.com/debris/bignumber.js.git/'
npm ERR! Linux 4.4.0-81-generic
npm ERR! argv "/home/travis/.nvm/versions/node/v7.8.0/bin/node" "/home/travis/.nvm/versions/node/v7.8.0/bin/npm" "install"
npm ERR! node v7.8.0
npm ERR! npm  v4.2.0
npm ERR! code 128
npm ERR! Command failed: git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror https://github.com/debris/bignumber.js.git /home/travis/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1
npm ERR! Cloning into bare repository '/home/travis/.npm/_git-remotes/git-https-github-com-debris-bignumber-js-git-master-df3f1ce1'...
npm ERR! remote: Invalid username or password.
npm ERR! fatal: Authentication failed for 'https://github.com/debris/bignumber.js.git/'
npm ERR! 
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm ERR! Please include the following file with any support request:
npm ERR!     /home/travis/.npm/_logs/2017-06-24T04_53_58_200Z-debug.log
The command "npm install" failed and exited with 1 during .
```

Looks to me like the repo got deleted...

https://github.com/debris/bignumber.js

wonder if this is compatible:

https://github.com/MikeMcl/bignumber.js/

**Comment by OR13** [comment]
This issue suggests that web3 may be the actual culprit...

https://github.com/ethereumjs/testrpc/issues/163

**Comment by OR13** [comment]
It seems web3.js is aware of the issue: https://github.com/ethereum/web3.js/issues/904

**Comment by OR13** [comment]
I gave this is shot, but could not get CI to pass, and the PR was starting to look to embarrassing to leave open. 

```
$ npm install web3@latest --save
$ npm install 
```

works for me locally. node v8.1.2

If anyone knows why CI fails for all these terrible commits: 

https://github.com/ethereumjs/testrpc/pull/333

I'd love to know how to not look this dumb in the future :)

**Comment by mohanbabubb** [comment]
Im struck with this failing dependency 'https://github.com/debris/bignumber.js.git/', i dont have any cache or mirrors to by pass. Any way to specify or install the above dependency manually.?
I see. @OR13  had a local cache which works for him; i'm assuming.!!

**Comment by yigitozkavci** [comment]
Can we do something like this to this repo as well? @OR13 https://github.com/ethereum/web3.js/pull/905 Because https://github.com/debris/bignumber.js.git/ no more exists and npm tries to fetch it from source, apparently.

**Comment by ghost** [comment]
I was having the same problem but managed to install the beta without problems:

`$ npm install -g ethereumjs-testrpc@beta`

**Comment by kowalski** [comment]
I'm affected too :(
The problem is caused by `"web3": "~0.16.0",` in `package.json`. It would go away if would go away if https://github.com/ethereumjs/testrpc/pull/309 is merged.

**Comment by ayinot** [comment]
Work around for this.

1. Clone "https://github.com/ethereumjs/testrpc" in any of your machine folder.
2. Open it in any editor, open the package.json
3. Change the repository of bignumber.js as 
    ......
    "bignumber.js": "git://github.com/MikeMcl/bignumber.js.git",
    ......
4. Remove web3, web3-provider-engine.
5. Run "npm install ."
6. Install web3, web3-provider-engine
7. Then "npm start"

cheers your ethereum-testrpc will get started.

**Comment by dylanseago** [comment]
I don't think https://github.com/ethereumjs/testrpc/pull/309 will resolve the issue because it depends on web3-provider-engine@~11.0.2 which also depends on web3@^0.16.0.

**Comment by danchay** [comment]
Thank you ToniyaSundaram.

**Comment by analogrytm** [comment]
Yep its still broken for ethereumjs-testrpc. But they just fixed the npm package for web3.

**Comment by ponarunkumar** [comment]
@ToniyaSundaram Your workaround solution did not work. When you get a chance Could you please share more details if you had the testrpc installed successfully?

**Comment by dylanseago** [comment]
@ponarunkumar Take a look at my PR https://github.com/ethereumjs/testrpc/pull/334. Tests are failing but it should work until this issue can be properly addressed.

**Comment by Magneticmagnum** [comment]
Thanks dylanseago, that worked for me. 👍

**Comment by ponarunkumar** [comment]
@dylanseago thanks for your reference PR #. However, upgrading the web3 dependency didn't help. I am not sure whether I am missing something else. @Magneticmagnum can you please elaborate on the steps which worked for you when you get a chance. The issue is still open and unable to install -g ethereumjs-testrpc

**Comment by rockytang** [comment]
same problem here with yarn.
error Refusing to download the git repo "https://github.com/debris/bignumber.js.git" over HTTPS without a commit hash - pos
```
yarn global add ethereumjs-testrpc
yarn global v0.21.3
warning No license field
[1/4] 🔍  Resolving packages...
error Refusing to download the git repo "https://github.com/debris/bignumber.js.git" over HTTPS without a commit hash - pos
sible certificate error?
info Visit https://yarnpkg.com/en/docs/cli/global for documentation about this command.
```
yarn version: 0.21.3
node version: v6.11.0

**Comment by rockytang** [comment]
Updating to the latest versions of bignumber.js, web3, and web3-provider-engine will fix this locally.
@ponarunkumar you can pull down this PR https://github.com/ethereumjs/testrpc/pull/335/files, and run `npm install`, `npm start` 
However, you will still not be able to do install -g until it's merged.

**Comment by dylanseago** [comment]
@rockytang I don't think the problem is related to the direct bignumber dependency, rather web3 used a github link to depend on bignumber which has since been deleted.

@ponarunkumar The following worked for me:
```
npm install -g web3@latest
npm install -g web3-provider-engine@latest
git clone https://github.com/dylanseago/testrpc.git -b web3-fix ethereumjs-testrpc
cd ethereumjs-testrpc
npm link web3
npm link web3-provider-engine
npm install -g
```
You may find that upgrading web3-provider-engine breaks some things but it's the best workaround I have so far.

**Comment by Magneticmagnum** [comment]
@rockytang Sorry, I wasn't specific.  Yes, it worked for a local install only.  Though dylan's instructions above will work as well if you need -g.

**Comment by ponarunkumar** [comment]
Thanks @rockytang appreciate your quick response! Yes, @dylanseago workaround worked fine now.

**Comment by tcoulter** [comment]
Hi all. 

Though this issue is unfortunate, we've been working very hard to complete remove this and all future installation issues from the TestRPC, as well as Truffle. Please see the following two blog posts for a description of what we're doing:

- http://truffleframework.com/blog/how-were-making-installation-issues-a-thing-of-the-past
- http://truffleframework.com/blog/removing-installation-issues-continued-testrpc

The TL;DR version of this is that you should use the `beta` versions of Truffle and the TestRPC. This will not only fix this specific installation issue, but most (if not all) installation issues, ever. We're likely to release these versions this week. 

Please keep us updated, and we appreciate your feedback.

**Comment by tcoulter** [comment]
`beta` has been officially released to production. Closing this issue for housekeeping. Thanks all!
